exporters/otlp/otlpmetric/internal/exporter.go

```diff
@@ -96,8 +93,11 @@ func (e *exporter) Shutdown(ctx context.Context) error {
 // New return an Exporter that uses client to transmits the OTLP data it
 // produces. The client is assumed to be fully started and able to communicate
 // with its OTLP receiving endpoint.
-func New(client Client) metric.Exporter {
-	return &exporter{client: client}
+func New(client Client, cs ConfigSelector) metric.Exporter {
```

Comment:
This change would make the `otlpmetric` package not backwards compatible.  This shouldn't happen, but there is the case where someone upgrades grpc without upgrading the HTTP, it would pull this package up and break the not upgraded one.  See https://github.com/open-telemetry/opentelemetry-go/issues/3846 for breaking this dependency.

Can we leave the arguments to New as is and use a different function to accomplish this?

That will mean that New() will need some way to get the ConfigSelector.